tests/integration-e2e/run-e2e.sh

```
@@ -0,0 +1,190 @@
+#
+# Phase 1: bring up the DB + control plane (platform-api), then via the
+#          platform-api REST API create a project, a REST API (upstream =
+#          sample-backend), a gateway and a registration token.
+# Phase 2: start the gateway data plane (gateway-controller + gateway-runtime)
+#          with that token, deploy the API, and assert a request through the
+#          gateway ingress reaches the sample backend.
+#
+# Usage: ./run-e2e.sh [up|down]   (default: full run + teardown)
+set -uo pipefail
+
+cd "$(dirname "$0")"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)